### PR TITLE
NXP S32K1XX: add initial support for NXP RDDRONE-BMS772 board

### PR DESCRIPTION
**Pull request by jarivanewijk**
## Summary
Adds initial support for the NXP RDDRONE-BMS772 board. The basic NSH configuration just initializes the board and enables the console UART. Code to enable other peripherals is also included, but remain mostly untested for the time being.

## Impact
Adds a new board configuration for NXP RDDRONE-BMS772 with NXP S32K144 MCU.

## Testing
Basic NSH config tested on RDDRONE-BMS772 hardware. Most other functionality (not enabled in the :nsh config) has also been tested, but with an older NuttX release. We first need to update our BMS application before we can again test complete board functionality.

### CODE REVIEW

File: boards/arm/s32k1xx/rddrone-bms772/Kconfig [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+menu "PM configuration for PM_STANDBY mode"
+
+choice
+	prompt "MCU power mode"
+	default VLPR_STANDBY
+
+config RUN_STANDBY
+	bool "RUN mode"
+	---help---
+		The MCU should be in RUN mode in the PM_STANDBY mode
```
Comment (acassis) [inline boards/arm/s32k1xx/rddrone-bms772/Kconfig:39]: Maybe the description name of this mode should be more clear: i.e.: "Keep in RUN mode during PM Standby"

File: boards/arm/s32k1xx/rddrone-bms772/Kconfig [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+endmenu # PM configuration for PM_STANDBY mode
+
+menu "PM configuration for PM_SLEEP mode"
+
+choice
+	prompt "MCU power mode"
+	default VLPR_SLEEP
+
+config RUN_SLEEP
+	bool "RUN mode"
```
Comment (acassis) [inline boards/arm/s32k1xx/rddrone-bms772/Kconfig:113]: Same here: "Keep in RUN mode during PM Sleep mode"

File: boards/arm/s32k1xx/rddrone-bms772/README.txt [hunk 3]
```diff
@@ -0,0 +1,142 @@
+README
```
Comment (acassis) [inline boards/arm/s32k1xx/rddrone-bms772/README.txt:1]: Please add the board Documentation too: https://nuttx.apache.org/docs/latest/platforms/index.html

File: boards/arm/s32k1xx/rddrone-bms772/Kconfig (hunk 1 above)
Comment (jarivanewijk) [reply boards/arm/s32k1xx/rddrone-bms772/Kconfig]: I agree. I changed the description and help text. We plan to revisit all of these PM-related options in the near future anyway, as it is very difficult to understand this configuration for a new user.

File: boards/arm/s32k1xx/rddrone-bms772/README.txt (hunk 3 above)
Comment (acassis) [reply boards/arm/s32k1xx/rddrone-bms772/README.txt:1]: Ok! Thank you!

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**